source/overloading.tex

```diff
@@ -1794,10 +1794,10 @@
 \begin{example}
 \begin{codeblock}
 struct S {
-  auto operator<=>(const S&, const S&) = default; // \#1
-  bool operator<(const S&, const S&);             // \#2
+  std::strong_ordering operator<=>(const S&) const;  // \#1
+  bool operator<(const S&) const;                    // \#2
```

Comment:
Why did you choose this avenue, instead of adding "friend"?
In particular, why did you change "auto" to std::strong_ordering" here?

Reply:
1: I was trying not to deviate from what I believe to be Herb's preferred syntax, `auto operator<=>(const S&) const = default;`.
2: But in this case, the fact that this operator is *defaulted* is not relevant to the example, so I removed the `= default`.
3: But without an `= default`, the use of `auto` is at least weird (but I guess not actually ill-formed, as long as the function definition also uses `auto`). So I replaced `auto` with `strong_ordering`.

Pulling on this thread probably wasn't worth it. I think I'll stop trying to use `operator<=>(const S&) const` and go back to my rule that all operators should be ADL friend functions. It makes life so much simpler, even if it is more verbose. :)